flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -315,6 +322,7 @@ public void returnExclusiveSegments(List<MemorySegment> segments) throws IOExcep
 	public void setInputChannel(IntermediateResultPartitionID partitionId, InputChannel inputChannel) {
 		synchronized (requestLock) {
 			if (inputChannels.put(checkNotNull(partitionId), checkNotNull(inputChannel)) == null
+					&& indexToInputChannelMap.put(inputChannel.getChannelIndex(), inputChannel) == null
```

Comment:
Please extract `inputChannels.put(...)` and `indexToInputChannelMap.put(...)` to separate method wrapping those calls - this logic is duplicated with `updateInputChannel(...)`.

Also change it to something like this:

```
private boolean addInputChannel(inputChannel, partitionId) {
  boolean alreadyInInputChannels = inputChannels.put(...) != null;
  boolean alreadyInIndexesMapping = indexToInputChannelMap.put(...) != null;
  checkState(alreadyInInputChannels ^ alreadyInIndexesMapping, "Inconsistent internal state (...)");
  return alreadyInInputChannels;
}
```

Reply:
alright.